### PR TITLE
v2 nginx adapter

### PR DESCRIPTION
**Pull request by kumarabd**
Signed-off-by: kumarabd <abishekkumar92@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: nginx/sample_apps.go [hunk 1]
```diff
@@ -1,29 +1,116 @@
...
+func (nginx *Nginx) installSampleApp(namespace string, del bool, templates []adapter.Template) (string, error) {
 	st := status.Installing
 
 	if del {
 		st = status.Removing
 	}
 
 	for _, template := range templates {
-		contents, err := utils.ReadRemoteFile(string(template))
+		contents, err := readFileSource(string(template))
```
Comment (utkarsh-pro) [inline nginx/sample_apps.go:23]: @kumarabd I think we have this function in meshkit now :sweat_smile:

File: internal/config/config.go [hunk 2]
```diff
@@ -84,38 +89,38 @@ func NewKubeconfigBuilder(provider string) (config.Handler, error) {
...
+	if err = installBinary(NginxExecutable, runtime.GOOS, res); err != nil {
+		return err
 	}
 
-	fmt.Println("Done")
-	return path.Join(configRootPath, "nginx-meshctl"), nil
+	return nil
 }
 
 func downloadBinary(platform string) (*http.Response, error) {
```
Comment (utkarsh-pro) [inline internal/config/config.go:126]: downloadBinary was actually failing after a few days I created #17 . I believe the reason for that is that the urls have changed now. Can we instead place the binaries manually?

File: nginx/sample_apps.go (hunk 1 above)
Comment (kumarabd) [reply nginx/sample_apps.go:23]: Oh that's great then, kill change them

**Review by leecalcote — COMMENTED** [review]
Is this PR antiquated? Ready for deletion?

File: internal/config/error.go [hunk 3]
```diff
@@ -19,9 +19,14 @@ import (
...
 const (
-	ErrEmptyConfigCode = "11300"
+	ErrEmptyConfigCode   = "test"
+	ErrInstallBinaryCode = "test"
 )
 
 var (
 	ErrEmptyConfig = errors.NewDefault(ErrEmptyConfigCode, "Config is empty")
 )
+
```
Comment (leecalcote) [inline internal/config/error.go:29]:
```suggestion
// ErrInstallBinary captures failure to update filesystem permissions
```

File: nginx/nginx.go [hunk 4]
```diff
@@ -14,26 +14,16 @@ import (
 
 type Nginx struct {
```
Comment (leecalcote) [inline nginx/nginx.go:15]:
```suggestion
// Nginx defines a model for this adapter
type Nginx struct {
```

File: nginx/error.go [hunk 5]
```diff
@@ -1,54 +1,58 @@
...
 
 var (
-	ErrInstallNginxCode = "nginx_test_code"
-	ErrMeshConfigCode   = "nginx_test_code"
-	ErrClientConfigCode = "nginx_test_code"
-	ErrStreamEventCode  = "nginx_test_code"
-	ErrExecDeployCode   = "nginx_test_code"
-	ErrExecRemoveCode   = "nginx_test_code"
-	ErrSampleAppCode    = "nginx_test_code"
+	ErrInstallNginxCode    = "nginx_test_code"
```
Comment (leecalcote) [inline nginx/error.go:8]:
```suggestion
        // ErrInstallNginxCode should really have an error code defined by now.
	ErrInstallNginxCode    = "nginx_test_code"
```

**Review by leecalcote — APPROVED** [review]

File: internal/config/config.go (hunk 2 above)
Comment (kumarabd) [reply internal/config/config.go:126]: @utkarsh-pro the binary is already in place. this is just for in-case scenarios.

**Review by leecalcote — APPROVED** [review]

**Pull request merged**